subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/typeHandling/inMemory/PersistedBytes.java

```diff
@@ -5,6 +5,7 @@
 
 import java.nio.ByteBuffer;
 
+@SuppressWarnings({"PMD.ArrayIsStoredDirectly", "PMD.MethodReturnsInternalArray"})
```

Comment:
why are we suppressing these here?

Reply:
code was written 2008, does it really make sense to rewrite it, and clone the bytes? if something is working for so long imo it is better to focus on newly introduced vulnerabilites.

Reply:
The age of code does not render it unworthy of refactoring - on the contrary: legacy code is more likely to need refactoring as it might be using outdated language features or programming patterns. The fact that it works for so long is irrelevant, as we're not talking about fixing it, but about refactoring it to improve its quality.
Also, considering that we have performance issues for a long time, I don't expect them to originate mainly from newly introduced code.

We can configure qualitative scans in a way that only newly introduced flaws block the pipeline, so there's no urgent need to suppress potentially valid complaints by our scan tools. We should rather address them in dedicated efforts to improve our code (especially the ages-old areas).

If the code's age is the main or even only reason for suppressing it, I don't think we should suppress.

Reply:
the age of the code is not the only reason to suppress. main reason is if there are 100 warnings, and one is added, it is difficult to spot, and it will become 101. if the old ones are suppressed there is 0 warnings, and it will stay like it. not everything should be suppressed of course - ideally it works like in this pull request that suppressions are properly challenged.

Reply:
if you really want to fix it or get it fixed, TODO is the better concept imo. should i add one?

Reply:
see here as well:
https://www.appsloveworld.com/java/100/140/reasoning-behind-arrayisstoreddirectly-rule-of-pmd

Reply:
took approach B).

Reply:
my bad, should've been more explicit... the TODO comment should reference a **GitHub** issue.
> B) suppress the warnings and "replace" them with a TODO comment referencing ~~an issue~~ **edit: a GitHub issue** that outlines the reasoning behind the rules and discusses possible mitigating steps including the performance-vs-security trade-off.